pkg/controller/multiarchbuildconfig/multiarchbuildconfig_test.go

```diff
@@ -25,41 +26,115 @@ import (
 	"github.com/openshift/ci-tools/pkg/manifestpusher"
 )
 
+var (
+	scheme *runtime.Scheme
+)
+
+type mockManifestPusher struct {
+	errToReturn error
+}
+
+func (m *mockManifestPusher) PushImageWithManifest(builds []buildv1.Build, targetImageRef string) error {
+	return m.errToReturn
+}
+
+type buildBuilder struct {
+	name     string
+	arch     string
+	phase    buildv1.BuildPhase
+	mabcName string
+}
+
+func NewBuildBuilder() *buildBuilder {
+	return &buildBuilder{mabcName: "foo"}
+}
+
+func (bb *buildBuilder) Name(name string) *buildBuilder {
+	bb.name = name
+	return bb
+}
+
+func (bb *buildBuilder) Arch(arch string) *buildBuilder {
+	bb.arch = arch
+	return bb
+}
+
+func (bb *buildBuilder) Phase(phase buildv1.BuildPhase) *buildBuilder {
+	bb.phase = phase
+	return bb
+}
+
+func (bb *buildBuilder) MABCName(mabcName string) *buildBuilder {
+	bb.mabcName = mabcName
+	return bb
+}
+
+func (bb *buildBuilder) Build() buildv1.Build {
+	return buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: bb.name,
+			Labels: map[string]string{
+				v1.MultiArchBuildConfigNameLabel: bb.mabcName,
+				v1.MultiArchBuildConfigArchLabel: bb.arch,
+			},
+		},
+		Status: buildv1.BuildStatus{Phase: bb.phase},
+	}
+}
+
+func TestMain(m *testing.M) {
```

Comment:
I don't see any point in having this test.

Reply:
It defines and configure a `scheme` that is common to a couple of test. Again, I can change this too if you prefer it to be declared separately.

Reply:
You can load the scheme in the init function. Can you try this instead?

Reply:
`TestMain` should be used to perform extra setup that is common to more than one tests and it runs only in tests, whereas `init` would always run. If you don't like `TestMain` I would rather setup the scheme in each test I need it than creating `init` in `multiarchbuildconfig_test.go`. Does it sound reasonable to you?

Reply:
There is no point of running a TestMain. If you need an higher layer of tests, you can create integration tests. Running the testMain just for load the schema is also no needed and its overkill, since you can do it with the init.